src/Components/CommandPrompt/CommandPrompt.tsx

```diff
@@ -0,0 +1,175 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+
+import styles from "./CommandPrompt.module.sass";
+import Layout from "Components/Layout/Layout";
+import Symbol from "Components/Symbol/Symbol";
+import { COMMAND_LIST } from "../../commands";
+import { Command } from "Models/Command";
+import Overlay from "Components/Overlay/Overlay";
+
+function CommandPrompt() {
+    const inputField = React.createRef<HTMLInputElement>();
+
+    const [shown, setShown] = useState<boolean>(false);
+    const [typed, setTyped] = useState("");
+    const [selectedCommandIndex, setSelectedCommandIndex] = useState(null);
+    const [matchedCommands, setMatchedCommands] = useState(COMMAND_LIST);
+    const commandToRun = useRef<Command>(null);
+
+
+    useEffect(() => {
+        setMatchedCommands(COMMAND_LIST.filter((command) => {
+            const lowerCommand = command.name.toLowerCase();
+            const lowerTyped = typed.toLowerCase();
+            return lowerCommand.includes(lowerTyped);
+        }))
+        setSelectedCommandIndex(matchedCommands.length >= 0 ? 0 : -1);
+    }, [typed])
+
+    useEffect(() => {
+        if (shown) {
+            inputField.current.focus();
+            setSelectedCommandIndex(0);
+        } else {
+            inputField.current.blur();
+            setTyped("");
+            if (commandToRun.current) {
+                commandToRun.current.callback();
+                commandToRun.current = null;
+            }
+        }
+    }, [shown])
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleKeyDownInDocument, { capture: false });
+
+        return function cleanup () {
+            document.removeEventListener('keydown', handleKeyDownInDocument, { capture: false });
+        }
+    }, [shown, selectedCommandIndex])
+
+    const handleKeyDownInDocument = (event) => {
+        // TODO: Shortcut in a settings pane ?
+        if (event.ctrlKey && event.key == 'k') {
```

Comment:
For cmd+k compatibility
```suggestion
        if ((event.ctrlKey || event.metaKey) && event.key == 'k') {
```